base/compiler/tfuncs.jl

```diff
@@ -939,7 +939,7 @@ function _fieldtype_tfunc(@nospecialize(s), exact::Bool, @nospecialize(name))
         ft = ftypes[fld]
     end
 
-    exactft = exact || !has_free_typevars(ft)
+    exactft = exact || (!has_free_typevars(ft) && u.name !== Tuple.name)
```

Comment:
Doesn't seem specific to the typevar case?
```
julia> Core.Compiler.fieldtype_tfunc(Type{Tuple{Any}}, Const(1))
Const(Any, false)
```

Reply:
`fieldtype(Tuple{Any}, 1)` does return `Any`, so I think that result is ok.

Reply:
ah, right